### PR TITLE
feat: Generate CloudFormation and Terraform docs files via AVD docs generator

### PR DESCRIPTION
**Pull request by reedloden**
Rather than rely on manual creation of `CloudFormation.md` and `Terraform.md`, generate them the same way as `docs.md` via the `avd_generator` program.

### CODE REVIEW

File: cmd/avd_generator/main.go [hunk 1]
```diff
@@ -72,3 +104,37 @@ var docsMarkdownTemplate = `
 {{ end}}
 {{ end }}
 `
+
+var terraformMarkdownTemplate = `
+{{ .Resolution }}
+
+{{ if .Terraform.GoodExamples }}
+{{ range .Terraform.GoodExamples }}
+{{ "```hcl" }}
```
Comment (reedloden) [inline cmd/avd_generator/main.go:113]: This doesn't work. I can't figure out how to escape the backticks in the template.

File: cmd/avd_generator/main.go [hunk 2]
```diff
@@ -72,3 +104,37 @@ var docsMarkdownTemplate = `
...
+{{ . }}
+{{ "```" }}
```
Comment (reedloden) [inline cmd/avd_generator/main.go:115]: Same

File: cmd/avd_generator/main.go [hunk 3]
```diff
@@ -72,3 +104,37 @@ var docsMarkdownTemplate = `
...
+{{ end}}
+{{ end }}
+`
+
+var cloudformationMarkdownTemplate = `
+{{ .Resolution }}
+
+{{ if .CloudFormation.GoodExamples }}
+{{ range .CloudFormation.GoodExamples }}
+{{ "```yaml" }}
```
Comment (reedloden) [inline cmd/avd_generator/main.go:130]: Same

File: cmd/avd_generator/main.go [hunk 4]
```diff
@@ -72,3 +104,37 @@ var docsMarkdownTemplate = `
...
+{{ . }}
+{{ "```" }}
```
Comment (reedloden) [inline cmd/avd_generator/main.go:132]: Same

File: cmd/avd_generator/main.go [hunk 5]
```diff
@@ -72,3 +104,37 @@ var docsMarkdownTemplate = `
 {{ end}}
 {{ end }}
 `
+
+var terraformMarkdownTemplate = `
+{{ .Resolution }}
+
+{{ if .Terraform.GoodExamples }}
```
Comment (reedloden) [inline cmd/avd_generator/main.go:111]: Whitespace is a bit messy, so likely need to remove some lines using `-`.

**Comment by reedloden** [comment]
To be clear, this is not yet working properly. Just wanted to push this up for comments. Once it's working correctly, I'll regenerate all the files and include in this PR (unless that should be done as a separate PR after this change lands).

**Comment by owenrumney** [comment]
Hey @reedloden  - I didn't want to push directly to your branch so here is the result of some fiddling... thanks for doing this, needed doing

*Terraform Template*
```go
var terraformMarkdownTemplate = `
{{ .Resolution }}

{{ if .Terraform.GoodExamples }}{{ range .Terraform.GoodExamples }}` + "```hcl" + `{{ . }}
` + "```" + `
{{ end}}{{ end }}
{{ if .Terraform.Links }}#### Remediation Links{{ range .Terraform.Links }}
 - {{ . }}
{{ end}}{{ end }}
`
```

*CloudFormation Template*
```go
var cloudformationMarkdownTemplate = `
{{ .Resolution }}

{{ if .CloudFormation.GoodExamples }}{{ range .CloudFormation.GoodExamples }}` + "```yaml" + `{{ . }}
` + "```" + `
{{ end}}{{ end }}
{{ if .CloudFormation.Links }}#### Remediation Links{{ range .CloudFormation.Links }}
 - {{ . }}
{{ end}}{{ end }}
`
```

**Comment by reedloden** [comment]
@owenrumney Thanks! I had to make a slight tweak to the CloudFormation template, as the `---` kept getting put right next to the `yaml`, but otherwise, works great. I've updated my branch and also ran `make docs`. Let me know if you'd prefer the docs be handled in a separate PR.

**Comment by reedloden** [comment]
I do see some cases where the existing docs may have been slightly better. I'm a bit time-constrained right now to go through each one, but feel free to make any necessary changes and push to my branch.

**Comment by owenrumney** [comment]
Hey @reedloden  - I've been away on hols, I'll look at this today and just push it. Thanks for the efforts

**Pull request merged**